### ISSUE TITLE
Configure level of trust for container-network

### ISSUE THREAD

**Issue by hickeng** [vmware/vic#5426]
**Story**
As a viadmin I know whether a container-network should be trusted or not
As an application author I don't want unnecessary firewall complexity to deal with on trusted networks

**Detail**
From https://github.com/vmware/vic/issues/4872#issuecomment-306972999

The secure-by-default configuration conflicts with ease-of-use and network performance in environments where trust can be assumed. This issue is to provide a means of configuring firewall requirements on a per container-network level.
Initial proposal is to add a `--container-network-firewall=<name>|<level>` option to vic-machine that will apply at `docker create` or `docker network connect` time.
The firewall levels would be along the lines of:
* open - no firewall
* peers - open on all ports to source addresses configured via `--container-network-ip-range`
* exposed - ports defined via `Dockerfile:EXPOSE` or `--expose` will be open for inbound
* published - ports explicitly published via `-p`  or `-P` will be open for inbound
* outbound - outbound connections only (implied for open, exposed, and published)
* blocked - firewall blocks everything by default and the container process manages access via iptables

One alternative would be to allow combination configuration such as `--container-network-firewall=<name>:outbound+published`

**Acceptance**
- [ ] able to connect between containers on the same container network on arbitrary ports
- [ ] <refined acceptance criteria from @malikkal

**Comment by hickeng** [comment]
This is flagged as need info to ensure the solution meets @malikkal's needs before further decomposition.
Added to 1.2 to ensure we deliver the change before the next supported drop.

**Comment by malikkal** [comment]
Thanks @hickeng This would be a great help. 

#4872 4872#issuecomment-307307571

**Comment by hickeng** [comment]
@malikkal The rules are constructed from the cVM configuration on containerVM boot - the primary inputs being the networks the container is connected to and the published ports.

Currently this checks whether the interface is for a container-network - this would switch to checking if the network is trusted, with container networks being trusted by default.

As such the configuration the rules are constructed from are set at `create` and `network connect` time.

**Comment by hickeng** [comment]
Estimate is a coarse 5 - still decisions needed as to how specifically to present this at a `vic-machine` level

**Comment by willsherwood** [comment]
No other flags in `vic-machine` have the format `--flag=value` but rather `--flag value`. Considering that, we might want to change the flag from `--container-network-firewall=<name>|<level>` to `--container-network-firewall <name>|<level>`. 

We can use the standard approach that if there are spaces in the name, surround the flag parameter with quotes: i.e. `--container-network-firewall "some name:outbound" --otherflags`.

**Comment by hickeng** [comment]
@willsherwood The args package we're using doesn't care whether it's a space or =, you can use both.
All of the original docs for `vic-machine` used the --flag=value style when discussing it and I've stuck with that as it's generally less error prone.

**Comment by hickeng** [comment]
@willsherwood Please be aware of #5575 - it changes how iptables is invoked, although should be mostly transparent to you. Means you can call iptables interactively via:
```
# chroot /.tether /iptables -L
```

**Comment by emlin** [comment]
@willsherwood Just realize that you're adding new option into vic-machine create.
As our decision for vic-machine configure, any new introduced create option, should be supported in configure as well. In addition, the container-network is already supported configure command, to support it in configure should be easy, but you need to make sure the workflow from configuration back to command option works well.

So basically, the feature scope for this new option should includes followings functions as well.

- make sure the network-firewall option can be updated together with new added container-network in reconfigure
- show the firewall configuration correctly in vic-machine inspect conf
- have corresponding test cases for reconfigure and inspect command
